gcloud-java-compute/src/main/java/com/google/gcloud/compute/ResourceId.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for Google Compute Engine resource identities.
+ */
+public abstract class ResourceId implements Serializable {
+
+  static final String REGEX = ".*?projects/([^/]+)/";
+  private static final String BASE_URL = "https://www.googleapis.com/compute/v1/projects/";
```

Comment:
Fine as is for now but I think we can get that from the service options (without the "project/" suffix).
Maybe we should create an issue for it?